cts/cts-scheduler.in

```diff
@@ -389,6 +403,53 @@ function test_results {
     fi
 }
 
+
+if test -z "$single_test" && test "$num_jobs" -ne 1 \
+        && command -v parallel >/dev/null 2>&1; then
+    _ACC_FILE=()
+    _ACC_DESC=()
+    _ACC_ARGS=()
+    export -f do_test_inner error failed info normalize show_test
+    export base create_mode diff_opts failed io_dir test_cmd
+    export CRM_EX_OK CRM_EX_ERROR CRM_EX_NOT_INSTALLED \
+           CRM_EX_USAGE CRM_EX_NOINPUT
+    do_test() {
+        test "$num_jobs" -ne 0 || num_jobs=100%
+        test $# -gt 1 || test $# -lt 1 || test "$1" != "EOT" || {
+            parallel -j "$num_jobs" --halt soon,fail=100 --plus \
+              --tagstring '{= $_="\033[30;3".(++$::color%8)."m" =}' \
+              do_test_inner '{1}' \
+                            '{=2 $_="[".seq()."/".total_jobs()."] ".$_ =}' \
+                            '{3}' \
+                ::: "${_ACC_FILE[@]}" \
+                :::+ "${_ACC_DESC[@]}" \
+                :::+ "${_ACC_ARGS[@]}"
+            num_failed=$?
+            if test $num_failed -gt 100; then
+                error "More than 100 tests failed"
+            fi
```

Comment:
Since that's how preciseness of exit status works with `parallel`
works IIRC, so I though it's better to be explicit -- even if this high
number means something is seriously busted.

Reply:
Revisited: one doesn't want to see that upon SIGINT.